### PR TITLE
[RLMRealm] Throw an exception when trying to set schema version or migra...

### PR DESCRIPTION
**Pull request by segiddins**
...te a realm when that realm is already open

Closes https://github.com/realm/realm-cocoa/issues/1455

\c @alazier @tgoyne

### CODE REVIEW

File: Realm/RLMRealm.mm [hunk 1]
```diff
@@ -807,6 +807,10 @@ + (void)setDefaultRealmSchemaVersion:(NSUInteger)version withMigrationBlock:(RLM
 }
 
 + (void)setSchemaVersion:(NSUInteger)version forRealmAtPath:(NSString *)realmPath withMigrationBlock:(RLMMigrationBlock)block {
+    if (realmsAtPath(realmPath)) {
```
Comment (alazier) [inline Realm/RLMRealm.mm]: You want to check count of this object

Comment (tgoyne) [inline Realm/RLMRealm.mm]: This check doesn't work. `realmsAtPath` can return an empty array.

File: Realm/RLMRealm.mm [hunk 2]
```diff
@@ -850,6 +854,10 @@ + (NSError *)migrateRealmAtPath:(NSString *)realmPath encryptionKey:(NSData *)ke
 }
 
 + (NSError *)migrateRealmAtPath:(NSString *)realmPath key:(NSData *)key {
+    if (realmsAtPath(realmPath)) {
```
Comment (alazier) [inline Realm/RLMRealm.mm]: check count of this object

File: Realm/Tests/MigrationTests.mm [hunk 3]
```diff
@@ -102,6 +103,8 @@ - (void)testGetSchemaVersion {
         [RLMRealm defaultRealm];
     }
 
+    [RLMRealm resetRealmState];
```
Comment (alazier) [inline Realm/Tests/MigrationTests.mm]: we shouldn't have to call this - better way to clear the cache is to do previous operations in an autoreleasepool. This ensures the RLMRealm objects have actually been disposed of.

**Comment by tgoyne** [comment]
Having to call `resetRealmState` to get things out of the cache always indicates you have a bug somewhere.

**Comment by segiddins** [comment]
Updated

**Comment by alazier** [comment]
check when setting encryption key as well

**Comment by alazier** [comment]
otherwise looks fine

**Comment by alazier** [comment]
need a test though

**Comment by segiddins** [comment]
@alazier test coverage added

**Comment by alazier** [comment]
need to update the CHANGELOG

**Comment by segiddins** [comment]
CHANGELING entry added

**Comment by alazier** [comment]
:+1:

**Comment by tgoyne** [comment]
:+1:

**Comment by mclark4386** [comment]
Um... anyone else having issue with this in Swift since in swift you can't try/catch to handle this gracefully if the realm is already open?

**Pull request merged**